### DISCUSSION TITLE
How to wait for all tasks to finish executing

### DISCUSSION THREAD

**Discussion by Yurunsoft** [tokio-rs/tokio#5369] [Q&A]
Code:

```rs
use tokio::time::{sleep, Duration};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    tokio::spawn(async move {
        sleep(Duration::from_millis(1000)).await;
        println!("OK");
    });
    Ok(())
}
```

No output after running.

I don't want to use `join!` myself, because I have some logic that will be executed by calling `tokio::spawn()` directly.

I think there should be a way to wait for all tasks to finish executing, but I didn't find in the documentation.

**Comment by Darksonn** [comment]
Tokio doesn't provide a way to do that.

**Reply by Yurunsoft** [reply]
Is there a plan to support this feature?

**Reply by Darksonn** [reply]
Currently, no.
